### PR TITLE
[PyOV] Remove deprecated py api

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - Remove deprecated python api
 - Wait for https://github.com/openvinotoolkit/openvino/pull/21592

### Tickets:
 - 127808

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/pyopenvino.cpp [hunk 1]
```diff
@@ -208,7 +208,7 @@ PYBIND11_MODULE(_pyopenvino, m) {
     regmodule_graph_types(m);
     regclass_graph_Dimension(m);  // Dimension must be registered before PartialShape
     regclass_graph_Layout(m);
-    regclass_graph_Shape(m);
+    regclass_graph_Shape(m);f
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/pyopenvino/pyopenvino.cpp:211]: remove?

Comment (akuporos) [inline src/bindings/python/src/pyopenvino/pyopenvino.cpp:211]:
```suggestion
    regclass_graph_Shape(m);
```

Comment (akuporos) [reply src/bindings/python/src/pyopenvino/pyopenvino.cpp]: done

File: src/bindings/python/src/openvino/runtime/ie_api.py [hunk 2]
```diff
@@ -369,15 +339,7 @@ def __call__(
...
         :type share_outputs: bool, optional
-        :param shared_memory: Deprecated. Works like `share_inputs` mode.
-
-                              If not specified, function uses `share_inputs` value.
-
-                              Note: Will be removed in 2024.0 release!
-                              Note: This is keyword-only argument.
 
-                              Default value: None
-        :type shared_memory: bool, optional
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/ie_api.py:380]: It's not removed from parameters of function. Same for all other infer-functions.
https://github.com/openvinotoolkit/openvino/pull/21675/files#diff-10e55df1b8f3b6e1926cee550ab907b024b095f2b5f4d26efd3c7a97f08ecb58R276

**Review by jiwaszki — CHANGES_REQUESTED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/python/src/openvino/runtime/ie_api.py (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/src/openvino/runtime/ie_api.py:380]: removed

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**